src/probnum/diffeq/odesolver.py

```diff
@@ -19,8 +19,21 @@ def solve(self, steprule):
         steprule : :class:`StepRule`
             Step-size selection rule, e.g. constant steps or adaptive steps.
         """
+        times, rvs = [], []
+        for t, rv in self.solver_steps(steprule):
+            times.append(t)
+            rvs.append(rv)
+
+        odesol = self.rvlist_to_odesol(times=times, rvs=rvs)
+        odesol = self.postprocess(odesol)
+        return odesol
+
+    def solver_steps(self, steprule):
```

Comment:
since this is probably the only controversial point in this PR, please feel free to slaughter this choice of names here :)